kb/cluster-metrics-reference.md

```diff
@@ -0,0 +1,148 @@
+---
+slug: cluster-metrics-reference
+title: Cluster metrics reference
+tags:
+  - kb-article
+date: 2022-09-15T00:00:00Z
+---
+
+The Temporal Cluster reports a range of metrics to help operators get visibility into the Cluster’s performance and set up alerts.
+All metrics emitted by the Cluster are listed in [defs.go](https://github.com/temporalio/temporal/blob/master/common/metrics/defs.go).
+
+Temporal uses [tally](https://github.com/uber-go/tally) for reporting metrics, and supports multiple backends like Prometheus, Statsd, and M3db.
+
+The [dashboards repository](https://github.com/temporalio/dashboards) contains community-driven Grafana dashboards that can be used for monitoring Temporal Server and SDK metrics.
+You can use these as a reference to build your own dashboards.
+For any metrics that are missing in the dashboards, use [defs.go](https://github.com/temporalio/temporal/blob/master/common/metrics/defs.go) as a reference.
+
+Note that apart from these metrics emitted by the Cluster, you should also monitor infrastructure-specific metrics like CPU, memory, and network for all hosts that are running Temporal Cluster services.
+
+#### Enable Cluster metrics
+
+Temporal Cluster reports all metrics by default.
+To enable receiving these metrics, expose ports (for Statsd) or listen addresses (for Prometheus) in the configuration, to push the metrics in the Statsd or Prometheus format.
+
+The following example shows how to set up Prometheus endpoint for Cluster metrics in your configuration:
+
+```
+metrics:
+    prometheus:
+        timerType: {{ default .Env.PROMETHEUS_TIMER_TYPE "histogram" }}
+        listenAddress: "{{ .Env.PROMETHEUS_ENDPOINT }}" #
+```
+
+The following example shows how to set up Statsd endpoint for Cluster metrics in your configuration:
+
+```
+metrics:
+    statsd:
+        hostPort: {{ .Env.STATSD_ENDPOINT }}
+        prefix: "temporal"
+```
+
+Depending on how you deploy your services, you can also define different endpoints for each service.
+
```

Comment:
This actually belongs in the Cluster configuration reference ya ?

Reply:
Updated